### PR TITLE
React upgrade

### PR DESCRIPTION
**Pull request by cenkce**
We fixed some typing issues which are caused by new version of React and ReactDom.

#### How to test
Upgrade React version to latest

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -5,7 +5,6 @@
 // - Mac: $HOME/Library/Application Support/Code/User/settings.json
 {
   "tslint.enable": true,
-  "editor.formatOnSave": true,
```
Comment (vince-fugnitto) [inline .vscode/settings.json]: Please revert this change.

File: packages/core/src/browser/shell/tab-bar-toolbar.tsx [hunk 2]
```diff
@@ -106,7 +107,7 @@ export class TabBarToolbar extends ReactWidget {
     protected render(): React.ReactNode {
         return <React.Fragment>
             {this.renderMore()}
-            {[...this.inline.values()].map(item => TabBarToolbarItem.is(item) ? this.renderItem(item) : item.render(this.current))}
+            {[... this.inline.values()].map(item => TabBarToolbarItem.is(item) ? this.renderItem(item) : item.render(this.current))}
```
Comment (vince-fugnitto) [inline packages/core/src/browser/shell/tab-bar-toolbar.tsx]: Please revert unnecessary changes, it looks like your styling may be different than the projects and creating unnecessary changes.

**Comment by vince-fugnitto** [comment]
Unfortunately, it looks like the previous issues have not been resolved with the latest changes:
- there is a conflict with the file `tab-bar-toolbar.tsx`
- there are build errors caused by changes in the `yarn.lock` file
   - make sure to perform a fresh build and commit the `yarn.lock` changes in its entirety
- the pull request commits have not been squashed which means that the `sign-off` fails
   - in general, we prefer that a single commit be present in a pull request unless the commits are logically different. We aim to have full featured commits meaning that each commit in the project's history successfully builds and is feature complete

**Comment by vince-fugnitto** [comment]
@cenkce do you still plan on updating the pull-request?
I noticed that there are still unresolved issues regarding the sign-off, eca, squashing and conflicts.

**Comment by cenkce** [comment]
@vince-fugnitto yes but I work with an another project now. I'm going to try in February again, thanks.

**Comment by vince-fugnitto** [comment]
Thank you for the contribution!
The pull-request will be closed to keep the list of open pull-requests manageable.
If you still plan on updating the pull-request, please provide your changes and it can be re-opened.
